File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        // Create a map to store Roman numeral values
        java.util.HashMap<Character, Integer> romanMap = new java.util.HashMap<>();
        romanMap.put('I', 1);
        romanMap.put('V', 5);
        romanMap.put('X', 10);
        romanMap.put('L', 50);
        romanMap.put('C', 100);
        romanMap.put('D', 500);
        romanMap.put('M', 1000);
        
        int total = 0;
        int length = s.length();
        
        // Loop through each character except the last one
        for (int i = 0; i < length - 1; i++) {
            int currentValue = romanMap.get(s.charAt(i));
            int nextValue = romanMap.get(s.charAt(i + 1));
            
            // If current value is less than next value, subtract it
            if (currentValue < nextValue) {
                total -= currentValue;
            } else {
                // Otherwise, add it
                total += currentValue;
            }
        }
        
        // Always add the last character's value
        total += romanMap.get(s.charAt(length - 1));
        
        return total;
    }
}

File: 13-roman-to-integer/roman-to-integer.php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function romanToInt($s) {
        $romanMap = [
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000
        ];
        
        $result = 0;
        $length = strlen($s);
        
        for ($i = 0; $i < $length; $i++) {
            $currentValue = $romanMap[$s[$i]];
            
            // Check if this is a subtraction case
            if ($i < $length - 1) {
                $nextValue = $romanMap[$s[$i + 1]];
                
                // If current value is less than next value, it's a subtraction case
                if ($currentValue < $nextValue) {
                    $result -= $currentValue;
                    continue;
                }
            }
            
            // Normal case: add the value
            $result += $currentValue;
        }
        
        return $result;
    }
}